### PR TITLE
Fixing Targeting Issue + Tests

### PR DESCRIPTION
**Pull request by mrm9084**
# Description

Fixes an issue with Targeting where its results didn't match DotNet. DotNet is Little endien and unsigned.

Added test file to test all feature flags with expected results.

### CODE REVIEW

File: featuremanagement/_defaultfilters.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 from ._featurefilters import FeatureFilter
 
-FEATURE_FLAG_NAME_KEY = "name"
+FEATURE_FLAG_NAME_KEY = "feature_name"
```
Comment (amerjusupovic) [inline featuremanagement/_defaultfilters.py:14]: I'm not super familiar with this library, but why is this key getting changed? Did the `context` variable structure change?

File: featuremanagement/_defaultfilters.py [hunk 2]
```diff
@@ -82,14 +82,19 @@ def _is_targeted(context_id, rollout_percentage):
         if rollout_percentage == 100:
             return True
 
-        hashed_context_id = hashlib.sha256(context_id.encode()).hexdigest()
-        context_marker = abs(int(hashed_context_id, 16))
-        percentage = (context_marker / (2**256 - 1)) * 100
+        hashed_context_id = hashlib.sha256(context_id.encode()).digest()
 
+        byte_array = bytearray(hashed_context_id)
+        decimal_values = [int(byte) for byte in byte_array[:4]]
```
Comment (rossgrambo) [inline featuremanagement/_defaultfilters.py:88]: This line doesn't do anything as far as I can tell. We can just pass `byte_array[:4]` into the next line instead.

Comment (mrm9084) [reply featuremanagement/_defaultfilters.py:88]: I think that section got added as debug steps. Removing.

**Review by rossgrambo — APPROVED** [review]

File: featuremanagement/_defaultfilters.py (hunk 1 above)
Comment (amerjusupovic) [reply featuremanagement/_defaultfilters.py:14]: Gotcha, thanks

**Review by amerjusupovic — APPROVED** [review]

**Pull request merged**